src/sentry/rules/processing/delayed_processing.py

```diff
@@ -13,24 +21,198 @@
+                    data_and_groups.group_ids.update(rules_to_groups[rule.id])
+                # Otherwise, create the tuple containing the condition data and the
+                # set of group_ids that apply to the unique condition
+                else:
+                    condition_groups[unique_condition] = DataAndGroups(
+                        condition_data, rules_to_groups[rule.id]
+                    )
+    return condition_groups
+
+
+def get_condition_group_results(
+    condition_groups: dict[UniqueCondition, DataAndGroups],
+    project: Project,
+) -> dict[UniqueCondition, dict[int, int]] | None:
+    condition_group_results: dict[UniqueCondition, dict[int, int]] = {}
+    for unique_condition, (condition_data, group_ids) in condition_groups.items():
+        condition_cls = rules.get(unique_condition.cls_id)
+
+        if condition_cls is None:
+            logger.warning("Unregistered condition %r", unique_condition.cls_id)
+            return None
+
+        condition_inst = condition_cls(project=project, data=condition_data)
+        if not isinstance(condition_inst, BaseEventFrequencyCondition):
+            logger.warning("Unregistered condition %r", condition_cls.id)
+            return None
+
+        _, duration = condition_inst.intervals[unique_condition.interval]
+        comparison_interval = condition_inst.intervals[unique_condition.interval][1]
+        comparison_type = (
+            condition_data.get("comparisonType", ComparisonType.COUNT)
+            if condition_data
+            else ComparisonType.COUNT
+        )
+        result = safe_execute(
+            condition_inst.get_rate_bulk,
+            duration,
+            comparison_interval,
+            group_ids,
+            unique_condition.environment_id,
+            comparison_type,
+        )
+        condition_group_results[unique_condition] = result
+    return condition_group_results
+
+
+def get_rules_to_fire(
+    condition_group_results: dict[UniqueCondition, dict[int, int]],
+    rule_to_slow_conditions: DefaultDict[Rule, list[MutableMapping[str, str] | None]],
+    rules_to_groups: DefaultDict[int, set[int]],
+) -> DefaultDict[Rule, set[int]]:
+    rules_to_fire = defaultdict(set)
+    for alert_rule, slow_conditions in rule_to_slow_conditions.items():
+        for slow_condition in slow_conditions:
+            if slow_condition:
+                condition_id = slow_condition.get("id")
+                condition_interval = slow_condition.get("interval")
+                target_value = int(str(slow_condition.get("value")))
+                for condition_data, results in condition_group_results.items():
+                    if (
+                        alert_rule.environment_id == condition_data.environment_id
+                        and condition_id == condition_data.cls_id
+                        and condition_interval == condition_data.interval
+                    ):
```

Comment:
We need to take into account whether the rule is `ANY` or `ALL` here. When `ALL`, we should only add to `rules_to_fire` if all slow conditions pass. This logic works fine as is for `ANY`.

Might be easier as two pass - so iterate through all the rules and conditions, and find out which ones actually fired. Then for each of those rules, add the groups

Reply:
:eyes: I think this I'll address in a separate PR since it'll need more test coverage